### PR TITLE
zh: add document for optimistic mode sharding merge

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What is changed, added or deleted? (Required)

add document for optimistic mode sharding merge, can see https://github.com/pingcap/docs-dm/blob/xiang/optimistic_ddl/zh/feature-shard-merge-optimistic.md

TODO: add error handle

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v2.0 changes for now)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by yikeke** [comment]
I converted this PR to draft. You can click "ready for review" after it gets ready~ @WangXiangUSTC

**Comment by WangXiangUSTC** [comment]
> I converted this PR to draft. You can click "ready for review" after it gets ready~ @WangXiangUSTC

ok

File: zh/feature-shard-merge-pessimistic.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+---
```
Comment (WangXiangUSTC) [inline zh/feature-shard-merge-pessimistic.md:1]: 内容来自于原来的 feature-shard-merge.md

**Comment by WangXiangUSTC** [comment]
@csuzhangxc PTAL

**Comment by csuzhangxc** [comment]
How about we don't cherry-pick it to release-1.0?

**Comment by WangXiangUSTC** [comment]
> How about we don't cherry-pick it to release-1.0?

OK

**Comment by WangXiangUSTC** [comment]
@toutdesuite Please remove the label `needs cherry-pick 1.0`

File: zh/skip-or-replace-abnormal-sql-statements.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ category: reference
 - `--sharding` 仅用于对 sharding group 预设一些操作，并且必须在 DDL 语句执行之前预设，不能在 DDL 语句已经执行后预设。
 
     - `--sharding` 模式下只支持预设，并只能使用 `--sql-pattern` 来匹配 binlog event。
-    - 有关使用 DM 处理 sharding DDL 同步的原理，请参阅[分库分表合并同步原理](feature-shard-merge.md#实现原理)。
+    - 有关使用 DM 处理 sharding DDL 同步的原理，[悲观模式分库分表合并同步原理](feature-shard-merge-pessimistic.md#实现原理)和[乐观模式分库分表合并同步原理](feature-shard-merge-optimistic.md#原理)。
```
Comment (csuzhangxc) [inline zh/skip-or-replace-abnormal-sql-statements.md:41]:
```suggestion
    - 有关使用 DM 处理 sharding DDL 同步的原理，请参阅[悲观模式分库分表合并同步原理](feature-shard-merge-pessimistic.md#实现原理)和[乐观模式分库分表合并同步原理](feature-shard-merge-optimistic.md#原理)。
```

File: zh/overview.md [hunk 3]
```diff
@@ -66,7 +66,7 @@ dmctl 是用来控制 DM 集群的命令行工具。
 
 ### Shard support
 
-DM 支持对原分库分表进行合库合表操作，但需要满足一些[使用限制](feature-shard-merge.md#使用限制)。
+DM 支持对原分库分表进行合库合表操作，但需要满足一些使用限制，详细信息请参考 [悲观模式分库分表合并同步使用限制](feature-shard-merge-pessimistic.md#使用限制)和[乐观模式分库分表合并同步使用限制](feature-shard-merge-optimistic.md#使用限制)。
```
Comment (csuzhangxc) [inline zh/overview.md:69]:
```suggestion
DM 支持对原分库分表进行合库合表操作，但需要满足一些使用限制，详细信息请参考[悲观模式分库分表合并同步使用限制](feature-shard-merge-pessimistic.md#使用限制)和[乐观模式分库分表合并同步使用限制](feature-shard-merge-optimistic.md#使用限制)。
```

File: zh/overview.md [hunk 4]
```diff
@@ -92,7 +92,7 @@ DM 支持对原分库分表进行合库合表操作，但需要满足一些[使
 + 分库分表
 
     - 如果业务分库分表之间存在数据冲突，可以参考[自增主键冲突处理](shard-merge-best-practices.md#自增主键冲突处理)来解决；否则不推荐使用 DM 进行同步，如果进行同步则有冲突的数据会相互覆盖造成数据丢失。
-    - 关于分库分表合并场景的其它限制，参见[使用限制](feature-shard-merge.md#使用限制)。
+    - 关于分库分表合并场景的其它限制，参见[使用限制](feature-shard-merge-pessimistic.md#使用限制)。
```
Comment (csuzhangxc) [inline zh/overview.md:95]: 要链接到乐观模式吗？

File: zh/feature-shard-merge.md [hunk 5]
```diff
@@ -5,173 +5,32 @@ category: reference
...
-综上可知，DM 在处理 sharding DDL 同步时，主要通过两级 sharding group 来进行协调控制，简化的流程为：
-
-1. 各 DM-worker 独立地协调对应上游 MySQL 实例内多个分表组成的 sharding group 的 DDL 同步。
-
-2. 当 DM-worker 收到所有分表的 DDL 语句时，向 DM-master 发送 DDL 相关信息。
+> **注意：**
+>
+> - 要执行分库分表合并同步任务，必须在任务配置文件中设置 `is-sharding: true`。
+> - DM 对分库分表的合并默认使用悲观协调模式，在文档中如果没特殊说明则默认为悲观协调模式。
+> - 在没有深入了解乐观模式的原理和使用限制的情况下不建议使用该模式，否则可能造成同步中断甚至数据不一致的严重后果。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge.md:18]: 2.0 我们换成让用户配置 `shard-mode` 吗？可能配置相关的文档也需要更新？

File: zh/feature-shard-merge.md [hunk 6]
```diff
@@ -5,173 +5,32 @@ category: reference
...
+### 悲观协调模式与乐观协调模式的对比
 
-8. 在完成被忽略的 DML 语句的重新同步后，所有 DM-worker 继续正常同步。
+| :----------- | :----------- |
+| 悲观协调模式   | 乐观协调模式   |
+| :----------- | :----------- |
+| 发起 DDL 的分表会暂停 DML 同步 | 发起 DDL 的分表会继续 DML 同步 |
+| 每个分表的 DDL 执行次序和语句必须相同 | 每个分表只需保持表结构互相兼容即可 |
+| DDL 在整个分表群达成一致后才同步到下游 | 每个分表的 DDL 会即时影响下游 |
+| 错误的 DDL 操作在侦测到后可以被拦截 | 错误的 DDL 操作会也被同步到下游，可能在侦测到之前已使部分上下游数据不一致 |
```
Comment (csuzhangxc) [inline zh/feature-shard-merge.md:36]:
```suggestion
| 错误的 DDL 操作在侦测到后可以被拦截 | 错误的 DDL 操作也会被同步到下游，可能在侦测到之前已使部分上下游数据不一致 |
```

File: zh/feature-shard-merge.md [hunk 7]
```diff
@@ -5,173 +5,32 @@ category: reference
...
+### 乐观协调模式
 
-6. 当 DM-master 收到 owner 执行 DDL 成功的消息后，请求其他所有 DM-worker 继续开始同步。
+在一个分表上执行的 DDL，DM 会自动修改成兼容其他分表的语句，并立即同步到下游，不会阻挡任何分表 DML 的同步。“乐观协调”模式的优点是处理 DDL 时不会阻塞数据的同步，但是使用不当会有同步中断甚至数据不一致的风险，详细的介绍请参考[乐观模式下分库分表合并同步](feature-shard-merge-optimistic.md)。
 
-7. 其他所有 DM-worker 各自开始重新同步在内部协调 DDL 同步过程中被忽略的 DML 语句。
+### 悲观协调模式与乐观协调模式的对比
 
-8. 在完成被忽略的 DML 语句的重新同步后，所有 DM-worker 继续正常同步。
+| :----------- | :----------- |
```
Comment (csuzhangxc) [inline zh/feature-shard-merge.md:30]: 这行是多余的，也可以看渲染后的 HTML。

File: zh/feature-shard-merge-optimistic.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+本文介绍了 DM 提供的乐观模式下分库分表的合并同步功能，此功能可用于将上游 MySQL/MariaDB 实例中结构相同/不同的表同步到下游 TiDB 的同一个表中。
+
+> **注意：**
+>
+> 在没有深入了解乐观模式的原理和使用限制的情况下不建议使用该模式，否则可能造成同步中断甚至数据不一致的严重后果。
+
+## 背景
+
+DM 支持在线上执行分库分表的 DDL 语句（通称 Sharding DDL），默认使用“悲观模式”，即当上游一个分表执行某一 DDL 后，这个分表的同步会暂停，等待其他所有分表都执行了同样的 DDL 才在下游执行该 DDL 并继续数据同步。
+这种“悲观协调”模式的优点是可以保证同步到下游的数据不会出错，问题是会中断同步进行不利于灰度。有些用户可能会花较长时间在单一分表执行 DDL，验证一定时间后后才会更改其他分表的结构。在悲观模式同步的设定下，这些 DDL 会阻塞同步，binlog 事件会大量积压。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:17]: 这个和 L16 是要换行的吗？

Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:17]:
```suggestion
这种“悲观协调”模式的优点是可以保证同步到下游的数据不会出错，缺点是会暂停数据同步而不利于对上游进行灰度变更。有些用户可能会花较长时间在单一分表执行 DDL，验证一定时间后才会更改其他分表的结构。在悲观模式同步的设定下，这些 DDL 会阻塞同步，binlog 事件会大量积压。
```

File: zh/feature-shard-merge-optimistic.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+因此，需要提供一种新的“乐观协调”模式，在一个分表上执行的 DDL，自动修改成兼容其他分表的语句后，立即同步到下游，不会阻挡任何分表 DML 的同步。
+
+## 原理
+
+在“乐观协调”模式下，DM-worker 接收到来自上游的 DDL 后，会把更新后的表结构转送给 DM-master。DM-worker 会追踪各分表当前的表结构，DM-master 合并成可兼容来自每个分表 DML 的合成结构，然后把与此对应的 DDL 同步到下游；对于 DML 会直接同步到下游。
+
+![optimistic-ddl-flow](/media/optimistic-ddl-flow.png)
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:25]: 感觉这里可能需要更新一下。我们实际实现时，向下游执行 DDL 是由 DM-worker 来做的，而不是 DM-master。

File: zh/feature-shard-merge-optimistic.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+### 例子
+
+例如上游 MySQL 有三个分表，使用 DM 同步到下游 TiDB，如下图所示：
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:29]:
```suggestion
例如上游 MySQL 有三个分表（`tbl00`, `tbl01` 以及 `tbl02`），使用 DM 同步到下游 TiDB 的 `tbl` 表中，如下图所示：
```

File: zh/feature-shard-merge-optimistic.md [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+
+在上游增加一列 `Level`：
+
+```SQL
+ALTER TABLE `tbl00` ADD COLUMN `Level` INT;
+```
+
+![optimistic-ddl-example-2](/media/optimistic-ddl-example-2.png)
+
+此时下游 TiDB 要准备接受来自 tbl00 有 Level 的 DML、以及来自 tbl01 和 tbl02 没有 Level 的 DML。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:41]:
```suggestion
此时下游 TiDB 要准备接受来自 `tbl00` 有 `Level` 的 DML、以及来自 `tbl01` 和 `tbl02` 没有 `Level` 的 DML。
```

File: zh/feature-shard-merge-optimistic.md [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+这时候如下的 DML 无需修改就可以同步到下游：
+
+```SQL
+UPDATE `tbl00` SET `Level` = 9 WHERE `ID` = 1;
+INSERT INTO `tbl02` (`ID`, `Name`) VALUES (27, 'Tony');
+```
+
+![optimistic-ddl-example-4](/media/optimistic-ddl-example-4.png)
+
+在 tbl01 同样增加一列 Level：
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:54]:
```suggestion
在 `tbl01` 同样增加一列 `Level`：
```

File: zh/feature-shard-merge-optimistic.md [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+
+```SQL
+ALTER TABLE `tbl01` ADD COLUMN `Level` INT;
+```
+
+![optimistic-ddl-example-5](/media/optimistic-ddl-example-5.png)
+
+此时下游已经有相同的 Level 列了，所以 DM-master 比较表结构之后不做任何操作。
+
+在 tbl01 刪除一列 Name：
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:64]:
```suggestion
在 `tbl01` 刪除一列 `Name`：
```

File: zh/feature-shard-merge-optimistic.md [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+
+```SQL
+ALTER TABLE `tbl01` DROP COLUMN `Name`;
+```
+
+![optimistic-ddl-example-6](/media/optimistic-ddl-example-6.png)
+
+此时下游仍需要接收来自 tbl00 和 tbl02 含 Name 的 DMLs，因此不会立刻删除该列。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:72]:
```suggestion
此时下游仍需要接收来自 `tbl00` 和 `tbl02` 含 `Name` 的 DMLs，因此不会立刻删除该列。
```

File: zh/feature-shard-merge-optimistic.md [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
+同样，各种 DML 仍可直接同步到下游：
+
+```SQL
+INSERT INTO `tbl01` (`ID`, `Level`) VALUES (15, 7);
+UPDATE `tbl00` SET `Level` = 5 WHERE `ID` = 5;
+```
+
+![optimistic-ddl-example-7](/media/optimistic-ddl-example-7.png)
+
+在 tbl02 增加一列 Level：
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:83]:
```suggestion
在 `tbl02` 增加一列 `Level`：
```

File: zh/feature-shard-merge-optimistic.md [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+
+```SQL
+ALTER TABLE `tbl02` ADD COLUMN `Level` INT;
+```
+
+![optimistic-ddl-example-8](/media/optimistic-ddl-example-8.png)
+
+此时所有分表都已有 Level 列。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:91]:
```suggestion
此时所有分表都已有 `Level` 列。
```

File: zh/feature-shard-merge-optimistic.md [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+
+在 tbl00 和 tbl02 各刪除一列 Name：
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:93]:
```suggestion
在 `tbl00` 和 `tbl02` 各刪除一列 `Name`：
```

File: zh/feature-shard-merge-optimistic.md [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+
+```SQL
+ALTER TABLE `tbl00` DROP COLUMN `Name`;
+ALTER TABLE `tbl02` DROP COLUMN `Name`;
+```
+
+![optimistic-ddl-example-9](/media/optimistic-ddl-example-9.png)
+
+到此步 Name 列也从所有分表消失了，所以可以安全从下游移除：
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:102]:
```suggestion
到此步 `Name` 列也从所有分表消失了，所以可以安全从下游移除：
```

File: zh/feature-shard-merge-optimistic.md [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+
+```SQL
+ALTER TABLE `tbl` DROP COLUMN `Name`;
+```
+
+![optimistic-ddl-example-10](/media/optimistic-ddl-example-10.png)
+
+## 风险 
+
+使用乐观同步时，由于 DDL 会即时同步到下游，若使用不当，可能导致上下游数据不一致。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:112]:
```suggestion
使用乐观模式同步时，由于 DDL 会即时同步到下游，若使用不当，可能导致上下游数据不一致。
```

File: zh/feature-shard-merge-optimistic.md [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+
+### 例子
+
+例如以下三个分表合并同步到 TiDB：
+
+![optimistic-ddl-fail-example-1](/media/optimistic-ddl-fail-example-1.png)
+
+在 tbl01 新增一列 Age，默认值定为 0：
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:120]:
```suggestion
在 `tbl01` 新增一列 `Age`，默认值定为 `0`：
```

File: zh/feature-shard-merge-optimistic.md [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+
+```SQL
+ALTER TABLE `tbl01` ADD COLUMN `Age` INT DEFAULT 0;
+```
+
+![optimistic-ddl-fail-example-2](/media/optimistic-ddl-fail-example-2.png)
+
+ 在 tbl00 新增一列 Age，但默认值定为 -1：
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:128]:
```suggestion
 在 `tbl00` 新增一列 `Age`，但默认值定为 `-1`：
```

File: zh/feature-shard-merge-optimistic.md [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+
+```SQL 
+ALTER TABLE `tbl00` ADD COLUMN `Age` INT DEFAULT -1;
+```
+
+![optimistic-ddl-fail-example-3](/media/optimistic-ddl-fail-example-3.png)
+
+此时所有来自 tbl00 的 Age 都不一致了。这是由于 `DEFAULT 0` 和 `DEFAULT -1` 互不兼容。虽然 DM 遇到这种情况会报错，但上下游不一致的问题就需要手动去解决。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:136]:
```suggestion
此时所有来自 `tbl00` 的 `Age` 都不一致了。这是由于 `DEFAULT 0` 和 `DEFAULT -1` 互不兼容。虽然 DM 遇到这种情况会报错，但上下游不一致的问题就需要手动去解决。
```

File: zh/feature-shard-merge-optimistic.md [hunk 23]
```diff
@@ -0,0 +1,169 @@
...
+
+- 执行每个批次的 DDL 前和后，要确保每个分表的结构达成一致。
+- 进行灰度 DDL 时，只集中在一个分表上测试。
+- 灰度完成后，在其他分表上尽量以最简单直接的 DDL 迁移到最终的 schema，而不要重放灰度测试中对或错的每一步。
+    - 例如：在分表执行过 `ADD COLUMN A INT; DROP COLUMN A; ADD COLUMN A FLOAT;`，在其他分表直接执行 `ADD COLUMN A FLOAT` 即可，不需要三条 DDL 都执行一遍。
+- 执行 DDL 时要注意观察 DM 同步状态。当同步报错时，需要判断这个批次的 DDL 是否会造成数据不一致。
+
+此外，不论是使用“乐观协调”或“悲观协调”，DM 仍是有以下限制：
+
+- 增量同步任务需要确认开始同步的 binlog position 对应的各分表的表结构必须一致。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:161]:
```suggestion
- 增量同步任务需要确保开始同步的 binlog position 对应的各分表的表结构必须一致。
```

File: zh/feature-shard-merge-optimistic.md [hunk 24]
```diff
@@ -0,0 +1,169 @@
...
+- 进入 sharding group 的新表必须与其他成员的表结构一致（正在执行一个 DDL 批次时禁止 `CREATE/RENAME TABLE`）。
+- 不支持 `DROP TABLE` / `DROP DATABASE`。
+- TiDB 不支持的 DDL 语句在 DM 也不支持。
+- 新增列的默认值不能包含 current_timestamp、rand()、uuid() 等，会造成上下游数据不一致。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:165]:
```suggestion
- 新增列的默认值不能包含 `current_timestamp`、`rand()`、`uuid()` 等，否则会造成上下游数据不一致。
```

File: zh/feature-shard-merge-optimistic.md [hunk 25]
```diff
@@ -0,0 +1,169 @@
...
+
+## 乐观协调模式的配置
+
+在任务的配置文件中指定 `shard-mode` 为 `optimistic` 则使用“乐观协调”模式，示例配置文件可以参考 [dm-task.yaml](https://github.com/pingcap/dm/blob/95d37b4f8cf36c5da84714f9c25d7502028bc835/tests/sequence_sharding_optimistic/conf/dm-task.yaml)。
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:169]: 更新并链接到 docs-dm 的配置文档怎么样？

File: zh/feature-shard-merge-optimistic.md (hunk 9 above)
Comment (WangXiangUSTC) [reply zh/feature-shard-merge-optimistic.md:24]: 这个图更新了

File: zh/feature-shard-merge.md (hunk 5 above)
Comment (WangXiangUSTC) [reply zh/feature-shard-merge.md:18]: 是指不需要配置 `is-sharding`了？

File: zh/feature-shard-merge-optimistic.md (hunk 25 above)
Comment (WangXiangUSTC) [reply zh/feature-shard-merge-optimistic.md]: 已改

**Comment by yikeke** [comment]
Please ping me if the technical review comments are all resolved. @WangXiangUSTC

File: zh/feature-shard-merge.md (hunk 5 above)
Comment (csuzhangxc) [reply zh/feature-shard-merge.md:18]: 嗯，`is-sharding` 只用来保持和 1.0 兼容

File: zh/feature-shard-merge-optimistic.md (hunk 8 above)
Comment (WangXiangUSTC) [reply zh/feature-shard-merge-optimistic.md]: 去掉了换行

File: zh/feature-shard-merge.md (hunk 7 above)
Comment (WangXiangUSTC) [reply zh/feature-shard-merge.md]: 删了

File: zh/feature-shard-merge.md (hunk 5 above)
Comment (WangXiangUSTC) [reply zh/feature-shard-merge.md:18]: 我改下

File: zh/feature-shard-merge-optimistic.md [hunk 26]
```diff
@@ -0,0 +1,168 @@
+---
+title: 乐观模式下分库分表合并同步
+category: reference
```
Comment (yikeke) [inline zh/feature-shard-merge-optimistic.md:3]:
```suggestion
summary: 介绍 DM 提供的乐观模式下分库分表的合并同步功能。
```
`category` meta is deprecated now.

File: zh/feature-shard-merge-optimistic.md [hunk 27]
```diff
@@ -0,0 +1,168 @@
...
+
+> **注意：**
+>
+> 在没有深入了解乐观模式的原理和使用限制的情况下不建议使用该模式，否则可能造成同步中断甚至数据不一致的严重后果。
+
+## 背景
+
+DM 支持在线上执行分库分表的 DDL 语句（通称 Sharding DDL），默认使用“悲观模式”，即当上游一个分表执行某一 DDL 后，这个分表的同步会暂停，等待其他所有分表都执行了同样的 DDL 才在下游执行该 DDL 并继续数据同步。这种“悲观协调”模式的优点是可以保证同步到下游的数据不会出错，缺点是会暂停数据同步而不利于对上游进行灰度变更。有些用户可能会花较长时间在单一分表执行 DDL，验证一定时间后才会更改其他分表的结构。在悲观模式同步的设定下，这些 DDL 会阻塞同步，binlog 事件会大量积压。
+
+因此，需要提供一种新的“乐观协调”模式，在一个分表上执行的 DDL，自动修改成兼容其他分表的语句后，立即同步到下游，不会阻挡任何分表 DML 的同步。
```
Comment (yikeke) [inline zh/feature-shard-merge-optimistic.md:18]:
```suggestion
因此，需要提供一种新的“乐观协调”模式，在一个分表上执行的 DDL，自动修改成兼容其他分表的语句后，立即同步到下游，不会阻挡任何分表执行的 DML 的同步。
```

File: zh/feature-shard-merge-optimistic.md [hunk 28]
```diff
@@ -0,0 +1,168 @@
...
+
+在 `tbl01` 刪除一列 `Name`：
+
+```SQL
+ALTER TABLE `tbl01` DROP COLUMN `Name`;
+```
+
+![optimistic-ddl-example-6](/media/optimistic-ddl-example-6.png)
+
+此时下游仍需要接收来自 `tbl00` 和 `tbl02` 含 `Name` 的 DMLs，因此不会立刻删除该列。
```
Comment (yikeke) [inline zh/feature-shard-merge-optimistic.md:71]:
```suggestion
此时下游仍需要接收来自 `tbl00` 和 `tbl02` 含 `Name` 的 DML 语句，因此不会立刻删除该列。
```

File: zh/feature-shard-merge-optimistic.md [hunk 29]
```diff
@@ -0,0 +1,168 @@
...
+
+- 增量同步任务需要确保开始同步的 binlog position 对应的各分表的表结构必须一致。
+- 进入 sharding group 的新表必须与其他成员的表结构一致（正在执行一个 DDL 批次时禁止 `CREATE/RENAME TABLE`）。
+- 不支持 `DROP TABLE` / `DROP DATABASE`。
+- TiDB 不支持的 DDL 语句在 DM 也不支持。
+- 新增列的默认值不能包含 `current_timestamp`、`rand()`、`uuid()` 等，否则会造成上下游数据不一致。
+
+## 乐观协调模式的配置
+
+在任务的配置文件中指定 `shard-mode` 为 `optimistic` 则使用“乐观协调”模式，示例配置文件可以参考 [DM 任务完整配置文件介绍](https://docs.pingcap.com/zh/tidb-data-migration/dev/task-configuration-file-full)。
```
Comment (yikeke) [inline zh/feature-shard-merge-optimistic.md:168]:
```suggestion
在任务的配置文件中指定 `shard-mode` 为 `optimistic` 则使用“乐观协调”模式，示例配置文件可以参考 [DM 任务完整配置文件介绍](/task-configuration-file-full.md)。
```
Use a relative link.

File: zh/feature-shard-merge-pessimistic.md [hunk 30]
```diff
@@ -0,0 +1,173 @@
...
+title: 悲观模式下分库分表合并同步
+category: reference
```
Comment (yikeke) [inline zh/feature-shard-merge-pessimistic.md:3]:
```suggestion
summary: 介绍 DM 提供的悲观模式（默认模式）下分库分表的合并同步功能。
```

File: zh/feature-shard-merge.md (hunk 5 above)
Comment (csuzhangxc) [reply zh/feature-shard-merge.md:18]: > 要执行分库分表合并同步任务，必须在任务配置文件中设置 `is-sharding: true`。

ping @WangXiangUSTC 这个要改一下吗

Comment (WangXiangUSTC) [reply zh/feature-shard-merge.md]: 去掉了 https://github.com/pingcap/docs-dm/pull/155/commits/82aae1f41606fbb11b0112e666b52db28e93180f

**Pull request merged**